salt/transport/zeromq.py

```diff
@@ -1280,14 +627,17 @@ def handle_future(future):
             send_timeout = self.io_loop.call_later(
```

Comment:
I've made changes to master.py so the PubServerChannel will connect early and the connection will be cached.

> Why would this code need grains? Should not all the information come from secure pillars or the conf files. Should transport just be establishing/maintaining transport protocol. It might add information into grains... e.g. minion id, however I would have thought the grain process would fetch this from a variable holding the minion id.
> 
> If grains are needed, maybe they should be processed before the transport is init.

We have many places where grains are getting pulled in but may not need to be. Unraveling all of this is out of the scope of this PR.

Reply:
This should be resolved since we are now caching the channels